File: python/1 - Iniciante/1118_varias_notas_com_validacao.py
numeros = []
contador = 0

while True:
    X = float(input())

    if 1 <= X <= 10:
        contador += 1
        numeros.append(X)

        if contador == 2:
            media = sum(numeros) / 2
            print(f'media = {media:.2f}')

            while True:
                a = input('novo calculo (1-sim 2-nao)\n')

                if a.isdigit():
                    a = int(a)

                    if a == 1:
                        numeros = []
                        contador = 0
                        break

                    elif a == 2:
                        exit()

    else:
        print(f'nota invalida')

File: c/c99/1 - Iniciante/1118_varias_notas_com_validacao.c
#include <stdio.h>

int main() {
    double N, soma, media;
    int divisor, X;

    while (1) {
        soma = 0;
        divisor = 0;

        while (divisor < 2) {
            scanf("%lf", &N);

            if (N < 0 || N > 10) {
                printf("nota invalida\n");
            } else {
                soma += N;
                divisor++;
            }
        }

        media = soma / divisor;
        printf("media = %.2lf\n", media);

        do {
            printf("novo calculo (1-sim 2-nao)\n");
            scanf("%d", &X);
        } while (X != 1 && X != 2);

        if (X == 2) {
            break;
        }
    }

    return 0;
}
